genesis-tools/contracts/vesting-contract/src/lib.rs

```diff
@@ -0,0 +1,515 @@
+        context.account_balance = to_yacto(1);
+
+        // Checking values in 1 day after genesis time
+        context.is_view = true;
+
+        for stake in &[1, 10, 100, 500, 999, 1001, 1005, 1100, 1500, 1999, 3000] {
+            let stake = *stake;
+            context.account_locked_balance = to_yacto(stake);
+            let balance_near = std::cmp::max(1000u128.saturating_sub(stake), 1);
+            context.account_balance = to_yacto(balance_near);
+            let extra_balance_near = stake + balance_near - 1000;
+
+            context.block_timestamp = to_ts(GENESIS_TIME_IN_DAYS + 1);
+            testing_env!(context.clone());
+
+            assert_eq!(contract.get_transferrable(), to_yacto(extra_balance_near));
+
+            // Checking values next day after cliff but before lockup timestamp
+            context.block_timestamp = to_ts(GENESIS_TIME_IN_DAYS - ALMOST_HALF_YEAR + YEAR + 1);
+            testing_env!(context.clone());
+
+            assert_eq!(contract.get_transferrable(), to_yacto(extra_balance_near));
+
+            // Checking values next day after lockup timestamp
+            context.block_timestamp = to_ts(GENESIS_TIME_IN_DAYS + YEAR + 1);
+            testing_env!(context.clone());
+
+            assert_almost_eq(contract.get_transferrable(), to_yacto(375 + extra_balance_near));
+
+            // Checking values middle of vesting
+            context.block_timestamp = to_ts(GENESIS_TIME_IN_DAYS - ALMOST_HALF_YEAR + YEAR * 2);
+            testing_env!(context.clone());
+
+            assert_almost_eq(contract.get_transferrable(), to_yacto(500 + extra_balance_near));
+
+            // Checking values a day before vesting ends
+            context.block_timestamp = to_ts(GENESIS_TIME_IN_DAYS - ALMOST_HALF_YEAR + YEAR * 4 - 1);
+            testing_env!(context.clone());
+
+            assert_almost_eq(contract.get_transferrable(), to_yacto(1000 + extra_balance_near));
+
+            // Checking values a day after vesting ends
+            context.block_timestamp = to_ts(GENESIS_TIME_IN_DAYS - ALMOST_HALF_YEAR + YEAR * 4 + 1);
+            testing_env!(context.clone());
+
+            assert_almost_eq(contract.get_transferrable(), to_yacto(1000 + extra_balance_near));
+        }
+    }
+
+    #[test]
+    fn test_transfer_call_by_owner() {
+        let (mut context, mut contract) = basic_setup();
+        context.block_timestamp = to_ts(GENESIS_TIME_IN_DAYS + YEAR + 1);
+        context.is_view = true;
+        testing_env!(context.clone());
+        assert_almost_eq(contract.get_transferrable(), to_yacto(375));
+
+        context.predecessor_account_id = account_owner();
+        context.signer_account_id = account_owner();
+        context.signer_account_pk = public_key(1);
+        context.is_view = false;
+        testing_env!(context.clone());
+
+        assert_eq!(env::account_balance(), to_yacto(LOCKUP_NEAR));
+        contract.transfer(to_yacto(100), non_owner());
```

Comment:
It would be good here to check that actually receipt with funds was created.
I also personally want to see an example of such check in unit tests @nearmax